### PR TITLE
UCS/VFS/LOG: Fix fork() child pid handling when assigning multiple atfork handlers

### PR DESCRIPTION
**Pull request by ofirfarjun7**
## What
Fix UCS logger wrongly handle fork() and prints wrong pid.

## Why ?
To reduce calls to getpid() process pid is cached on a static var. 
After calling to fork() process pid is changed and global vars are duplicated so the cached pid values of the child and parent are equal. 
UCS Logger handle that by attaching atfork handler to the child process. this handler change the log_pid var value to the child pid, 
but if the application attach more atfork handlers before UCS they will be called before UCS atfork handler and can cause the logger to run with the wrong pid...

From the Man:
"The handlers for each phase are called in a specified order: 
       the prepare handlers are called in reverse order of registration; 
       **the parent and child handlers are called in the order of registration**"

## How ?
By creating "init stage", as long as the init stage didn't complete UCS will call getpid() instead of using the cached value. Each time we call fork both child and parent will enter the init stage.

### CODE REVIEW

**Comment by ofirfarjun7** [comment]
TODO - add new application that attach multiple atfork handlers, use UCS and call fork to test the logger prints the correct pid.

**Review by yosefe — COMMENTED** [review]
LGTM besides adding unit test

File: src/ucs/debug/log.c [hunk 1]
```diff
@@ -91,6 +91,14 @@ static ucs_spinlock_t ucs_log_global_filter_lock;
 static khash_t(ucs_log_filter) ucs_log_global_filter;
 
 
+static inline int ucs_log_get_pid() {
```
Comment (yosefe) [inline src/ucs/debug/log.c:94]: code style

**Review by yosefe — APPROVED** [review]
pls squash

**Review by yosefe — APPROVED** [review]

**Pull request merged**